ui/src/wizard/AzureNameStep.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { isArray } from "util";
+import { IAzureNamingRules, IRelatedNameWizardContext } from "../../index";
+import { AzureWizardStep } from "./AzureWizardStep";
+
+export abstract class AzureNameStep<T extends IRelatedNameWizardContext> extends AzureWizardStep<T> {
+    protected abstract isRelatedNameAvailable(wizardContext: T, name: string): Promise<boolean>;
+
+    protected async generateRelatedName(wizardContext: T, name: string, namingRules: IAzureNamingRules | IAzureNamingRules[]): Promise<string | undefined> {
+        if (!isArray(namingRules)) {
+            namingRules = [namingRules];
+        }
+
+        let preferredName: string = namingRules.some((n: IAzureNamingRules) => !!n.lowercaseOnly) ? name.toLowerCase() : name;
+
+        for (const invalidCharsRegExp of namingRules.map((n: IAzureNamingRules) => n.invalidCharsRegExp)) {
+            preferredName = preferredName.replace(invalidCharsRegExp, '');
+        }
+
+        const minLength: number = Math.max(...namingRules.map((n: IAzureNamingRules) => n.minLength));
+        const maxLength: number = Math.min(...namingRules.map((n: IAzureNamingRules) => n.maxLength));
+
+        const maxTries: number = 1000;
```

Comment:
1000 seems excessive since each try is an Azure call

Reply:
... Or was this using the pre-fetched ones? If so, never mind.

Reply:
This is generic and its up to implementers on whether its an Azure call or whether is using pre-fetched ones. I can bring it down. How about 100?

Reply:
Sounds good for now.